modules/compute/gke-node-pool/main.tf

```diff
@@ -379,6 +393,18 @@ resource "google_container_node_pool" "node_pool" {
       condition     = var.enable_flex_start == true ? (var.spot == false) : true
       error_message = "Both enable_flex_start and spot consumption option cannot be set to true at the same time."
     }
+    precondition {
+      condition     = var.reservation_affinity.consume_reservation_type != "SPECIFIC_RESERVATION" || var.static_node_count <= local.reservation_available_count
+      error_message = "Requested static_node_count (${var.static_node_count}) exceeds the available reservation capacity (${local.reservation_available_count})."
+    }
+    precondition {
+      condition     = var.reservation_affinity.consume_reservation_type != "SPECIFIC_RESERVATION" || var.static_node_count == null || var.static_node_count > 0
+      error_message = "static_node_count must be greater than 0 or null when using specific reservations."
```

Comment:
Can't a user use auto scaling with specific reservation?

Reply:
I did not check the `null` condition mentioned. Given the null condition, this works with autoscaling as well.